lib/gridfs-stream/index.js

```diff
@@ -249,26 +276,25 @@ GridFSBucket.prototype.find = function(filter, options) {
  * @param {Number} [options.end] Optional 0-based offset in bytes to stop streaming before
  * @return {GridFSBucketReadStream}
  */
-
+const openDownloadStreamByNameSchema = {
+  revision: { type: 'number', default: -1 },
+  start: { type: 'number' },
+  end: { type: 'number' }
+};
 GridFSBucket.prototype.openDownloadStreamByName = function(filename, options) {
-  var sort = { uploadDate: -1 };
-  var skip = null;
-  if (options && options.revision != null) {
-    if (options.revision >= 0) {
-      sort = { uploadDate: 1 };
-      skip = options.revision;
-    } else {
-      skip = -options.revision - 1;
-    }
-  }
+  options = validate(
+    openDownloadStreamByNameSchema,
+    options,
+    {},
+    {
+      sort: options ? (options.revision >= 0 ? { uploadDate: 1 } : null) : null,
```

Comment:
I would like to avoid nested ternarys if possible. Can we make these calculations before the validate call?

Reply:
I can make these calculations before the validate call, but I thought they made the most sense here.